src/ui.rs

```diff
@@ -353,9 +353,17 @@ impl<UI: UserInterface> game::Loop<UI> for Loop<UI> {
                 input.update(input::Event::Mouse(mouse::Event::CursorTaken));
             }
 
-            window.update_cursor(new_cursor.into());
             self.mouse_cursor = new_cursor;
         }
+        // Use the game cursor if cursor is not on a UI element, use the mouse cursor otherwise
+        if self.mouse_cursor == MouseCursor::OutOfBounds {
+            let game_cursor = ui.cursor_icon();
+            window.set_cursor_visible(game_cursor != CursorIcon::Hidden);
+            window.update_cursor(game_cursor.into());
```

Comment:
Finally, if we make `CursorIcon` implement `TryFrom` instead of `From`, then we should be able to call `update_cursor` easily like this:

```rust
window.update_cursor(game_cursor.try_into().ok());
```

Reply:
Addressed in 66828daceee95b73aa10a45c2d279fe7600e3660.